File: cppXd/hannoi.cpp
#include <iostream>

using namespace std;

void show(int list)
void move (int lmt, int start[], int target[], int size){
    int I, J;
    for (int i = 0; i < size; i++){
        if (start[i] == lmt){
            start[i] = 0;
            I = i;
            break;
        }
    }

    for (int j = 0; j < size; j++){
        if (target[j] == 0){
            target[j] = lmt;
            J = j;
            break;
        }
    }

    cout<<lmt<<'['<<I<<"]->["<<J<<']'<<endl;
}

void set_interval(int value, int* start, int* aux, int* targ, int size){
    if (value == 1) move(value, start, targ, size);
    else{
        set_interval((value - 1), start, targ, aux, size);
        move(value, start, targ, size);
        set_interval((value - 1), aux, start, targ, size);
    }
}

int main(){
    int size;
    cout<<"Please enter the number of plates"<<endl;
    cin>>size;
    int start [size];
    int aux [size];
    int targ [size];

    for (int i = 0; i < size; i++) start[i] = i+1;
    cout<<"Beggining:";
    cout<<start<<"\n"<<aux<<"\n"<<targ<<endl;
    set_interval(size, start, aux, targ, size);
    cout<<"Finally:";
    cout<<start<<"\n"<<aux<<"\n"<<targ<<endl;

    return 0;
}

File: pythonXD/hannoi.py
import time


def move(lmt, start, target):
    size = len(start)
    for n in range(size):
        if start[n] == lmt:
            start[n] = 0
            break

    for m in range(size):
        if target[m] == 0:
            target[m] = lmt
            break

    print(f'{lmt}[{n}]->[{m}]')


def set_interval(value, start, aux, targ):
    if value == 1: move(value, start, targ)
    else:
        set_interval((value-1), start, targ, aux)
        move(value, start, targ)
        set_interval((value-1), aux, start, targ)


def main():
    size = int(input("Please enter the number of plates:\n"))
    start = [(size - n) for n in range(size)]
    temp = [0] * size
    targ = [0] * size
    print(f'\nBeggining:\n{start}\n{temp}\n{targ}')
    begin = time.time()
    set_interval(size, start, temp, targ)
    end = time.time()
    total = begin - end
    print(f"\nFinally:\n{start}\n{temp}\n{targ}")
    print(f"\nTotal time: {total}")


if __name__ == "__main__":
    main()
